OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbySavedSettingsBrowserLogic.cs

```diff
@@ -0,0 +1,351 @@
+		}
+
+		void Save()
+		{
+			var globalSettings = orderManager.LobbyInfo.GlobalSettings.Serialize();
+			var options = globalSettings.Value.ToDictionary()["Options"];
+			var unlockedOptions = options.Nodes.FindAll(optionYaml => optionYaml.Value.ToDictionary()["IsLocked"].Value == "False");
+			var optionsYaml = new List<MiniYamlNode>()
+			{
+				new MiniYamlNode(SettingsKeyMapId, map.Uid),
+				new MiniYamlNode(SettingsKeyOptions, new MiniYaml(options.Value, unlockedOptions))
+			};
+
+			if (!Directory.Exists(baseSavePath))
+				Directory.CreateDirectory(baseSavePath);
+
+			var filename = SettingsFilePath(saveTextField.Text);
+
+			Action inner = () =>
+			{
+				optionsYaml.WriteToFile(filename);
+				Ui.CloseWindow();
+			};
+
+			if (File.Exists(filename))
+			{
+				ConfirmationDialogs.ButtonPrompt(
+					title: "Overwrite saved settings?",
+					text: "Overwrite {0}?".F(saveTextField.Text),
+					onConfirm: inner,
+					confirmText: "Overwrite",
+					onCancel: () => { });
+			}
+			else
+				inner();
+		}
+
+		void Load()
+		{
+			var optionsYaml = MiniYaml.FromFile(selectedSave);
+
+			Ui.CloseWindow();
+
+			var savedMap = optionsYaml.Find(node => node.Key == SettingsKeyMapId).Value.Value;
+			if (savedMap != map.Uid)
+				orderManager.IssueOrder(Order.Command("map {0}".F(savedMap)));
+
+			var savedOptions = optionsYaml.Find(node => node.Key == SettingsKeyOptions);
+			foreach (var option in savedOptions.Value.Nodes)
+			{
+				string optionName = option.Key;
+				string optionValue = option.Value.ToDictionary()["Value"].Value;
+				orderManager.IssueOrder(Order.Command("option {0} {1}".F(optionName, optionValue)));
+			}
+		}
+
+		void Rename(string oldName, string newName)
+		{
+			try
+			{
+				var oldPath = SettingsFilePath(oldName);
+				var newPath = SettingsFilePath(newName);
+				File.Move(oldPath, newPath);
+
+				saves[saves.IndexOf(oldPath)] = newPath;
+				foreach (var c in saveList.Children)
+				{
+					var item = c as ScrollItemWidget;
+					if (item == null || item.ItemKey != oldPath)
+						continue;
+
+					item.ItemKey = newPath;
+					item.Get<LabelWidget>("TITLE").GetText = () => newName;
+				}
+
+				if (selectedSave == oldPath)
+					selectedSave = newPath;
+			}
+			catch (Exception ex)
+			{
+				Game.Debug("Failed to rename '{0}' to '{1}'. See the logs for details.", oldName, newName);
+				Log.Write("debug", ex.ToString());
+			}
+		}
+
+		void Delete(string savePath)
+		{
+			try
+			{
+				File.Delete(savePath);
+			}
+			catch (Exception ex)
+			{
+				Game.Debug("Failed to delete save file '{0}'. See the logs for details.", savePath);
+				Log.Write("debug", ex.ToString());
+				return;
+			}
+
+			if (savePath == selectedSave)
+				Select(null);
+
+			var item = saveList.Children
+				.Select(c => c as ScrollItemWidget)
+				.FirstOrDefault(c => c.ItemKey == savePath);
+
+			saveList.RemoveChild(item);
+			saves.Remove(savePath);
+		}
+
+		bool CanSave()
+		{
+			return saveTextField.Text != "";
```

Comment:
nit: No effective difference here, but a length check is more explicit IMO:
```suggestion
			return saveTextField.Text.Length != 0;
```